File: Year 2/Semester 4/MPP/SeventhWeekCSharpServerClient/persistence/interfaces/IEmployeeRepository.cs
using SeventhWeekCSharpServerClient.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SeventhWeekCSharpServerClient.repository.interfaces
{
    public interface IEmployeeRepository : IRepository<int, Employee<int>>
    {
        Employee<int> AuthenticateEmployee(string username, string password);
    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 5/JavaFifthWeek/src/main/java/com/example/temasaptamana5/repository/interfaces/IEmployeeRepository.java
package com.example.temasaptamana5.repository.interfaces;

import com.example.temasaptamana5.domain.Entity;
import com.example.temasaptamana5.repository.IRepository;

public interface IEmployeeRepository <ID, E extends Entity<ID>> extends IRepository<ID, E> {
}
